src/com/books/Main.java

```diff
@@ -0,0 +1,39 @@
+package com.books;
+
+import java.util.*;
+
+public class Main {
+    private static Set<Book> books;
+
+    public static void main(String[] args) {
+        books = new HashSet<Book>() {
+            @Override
+            public String toString() {
+                String result = "";
+                Iterator<Book> iterator = iterator();
+                while (iterator.hasNext()) {
+                    Book book = iterator.next();
+                    result += book + "\n";
+
+                }
+                return result;
+            }
+        };
+        Date date = new Date();
+
+        books.add(new Book("first book", date, new Person("Ivan", "First"), new Publisher("Publisher1")));
+        books.add(new Book("Second book", date, new Person("Andrew", "Second"), new Publisher("Publisher2")));
+        books.add(new Book("Third book", date, new Person("Sam", "Second"), new Publisher("Publisher3")));
+        System.out.println(books);
+
+    }
+
+    public static Set<Book> filerByAuthorName(String name) {
```

Comment:
Typo in "filter" word.
It's great that you decided to go forward and do next task, but just think about what happens with this method if we'll decide to move our application from console to the desktop application with user interface?